src/pages/signup-affiliates/index.tsx

```diff
@@ -86,7 +90,13 @@ const AffiliateSignup = () => {
 
     useEffect(() => {
         trackEvent({ action: 'open' })
-        return () => trackEvent({ action: 'close' })
+        const handleBeforeUnload = (event) => {
+            event.preventDefault()
+            event.returnValue = 'Are you sure you want to leave this page?'
+            trackEvent({ action: 'close' })
+        }
+        window.addEventListener('beforeunload', handleBeforeUnload)
+        return () => window.removeEventListener('beforeunload', handleBeforeUnload)
```

Comment:
@NikitK-deriv  won't this get triggered every time we move to another page?

Reply:
Nope, only when user want to close the tab. From this page we can only navigate back to partners page and in that particular case this action does not occur